recOrder/plugin/widget/main_widget.py

```diff
@@ -1539,6 +1540,12 @@ def connect_to_mm(self):
             self.mmc = bridge.get_core()
             self.mm = bridge.get_studio()
 
+            # Check MicroManager version compatibility.
+            bridge._master_socket.send({"command": "connect", "debug": False}) # latest versions of pycromanager use '_main_socket'
+            reply_json = bridge._master_socket.receive(timeout=500) # latest versions of pycromanager use '_main_socket'
+            if version.parse(reply_json['version']) < version.parse('4.0.0'):
+                print('WARNING: MicroManager version is incompatible with recOrder. Please upgrade to the latest tested MicroManager nightly build 20210713.')
+
```

Comment:
This checks if the MM ZMQ version is lower than 4.0.0. It's also possible to install a newer MM version and try to use it with recorder. We should add a check for ZMQ version grater than 4.0.0 and recommend downgrading to MM 20210713

Reply:
Good call. I've fixed this in the latest commits.